app/models/episode.rb

```diff
@@ -43,8 +43,9 @@ class Episode < BaseModel
 
   after_save :publish_updated, if: ->(e) { e.published_at_changed? }
 
-  scope :published,
-        -> { where('published_at IS NOT NULL AND published_at <= now()') }
+  scope :published, -> { where('published_at IS NOT NULL AND published_at <= now()') }
+
+  scope :published_by, -> (offset) { where('published_at IS NOT NULL AND published_at <= ?', Time.now + offset) }
```

Comment:
I left the current `published` scope, and added this new one to allow passing in a (positive or negative) offset